staging/src/k8s.io/apiserver/pkg/storage/value/transformer.go

```diff
@@ -126,7 +126,9 @@ func (t *prefixTransformers) TransformFromStorage(data []byte, context Context)
 	for i, transformer := range t.transformers {
 		if bytes.HasPrefix(data, transformer.Prefix) {
 			result, stale, err := transformer.Transformer.TransformFromStorage(data[len(transformer.Prefix):], context)
-			return result, stale || i != 0, err
+			if len(transformer.Prefix) != 0 || err == nil {
```

Comment:
can you invert this so it is a continue.  That's easier to read.  Generally exceptional cases are what we indent.  Also, add a comment about what the intent is.